deployment/constructs/dynamodb-table.ts

```diff
@@ -0,0 +1,77 @@
+import { Construct } from "constructs";
+import { RemovalPolicy } from "aws-cdk-lib";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
+
+interface DynamoDBTableProps {
+  readonly stage: string;
+  readonly isDev: boolean;
+  readonly name: string;
+  readonly partitionKey: { name: string; type: dynamodb.AttributeType };
+  readonly sortKey?: { name: string; type: dynamodb.AttributeType };
+  readonly lsi?: {
+    indexName: string;
+    sortKey: { name: string; type: dynamodb.AttributeType };
+  }[];
+  readonly gsi?: {
```

Comment:
I'd rip this out as it doesn't seem to be in use.

Reply:
Let's discuss in my response to another comment.  I don't think we should make our constructs app specific, that would create so many different versions/headaches.

Reply:
I'm fine with us making them expand over the course of this process of converting repos and then standardizing at the end with the most expansive file. 👍